security/fma-multi-threaded-64-bit-cannon.md

```diff
@@ -0,0 +1,164 @@
+
+### Insufficient memory in the program
+
+- **Description:** The op-program may run out of memory, causing it to crash.
+- **Risk Assessment:** High severity, low likelihood.
+- **Mitigations:** The 64-bit address space virtually eliminates memory exhaustion risks. Go's concurrent garbage collector automatically manages memory through scheduled background goroutines.
+- **Detection:** op-dispute-mon forecasts and alerts on undesirable game resolutions that would result due to a program crash.
+- **Recovery Path(s)**: See [Fault Proof Recovery](https://www.notion.so/oplabs/RB-000-Fault-Proofs-Recovery-Runbook-8dad0f1e6d4644c281b0e946c89f345f).
+
+### Compromised Control Flow in the program
+
+- **Description:** This could theoretically occur when the op-program runs out of memory in a way that lets the attacker reuse code to subvert execution.
+- **Risk Assessment:** High severity, low likelihood.
+  - Low likelihood: This requires an attacker to craft inputs that not only induce high memory usage, but also corrupt or spray the heap in a way that either produces invalid fault proofs or prevents valid fault proofs from being generated.
+- **Mitigations:** As with [Insufficient memory in the program](#insufficient-memory-in-the-program), the 64-bit address space effectively prevents this from occurring. Furthermore, the Go runtime checks memory allocations against heap corruption. However, such memory protections may not hold due to bugs in the Go runtime.
+- **Detection:** op-dispute-mon forecasts and alerts on undesirable game resolutions that would result due to honest claims being disputed at the bottom of the game tree.
+- **Recovery Path(s)**: See [Fault Proof Recovery](https://www.notion.so/oplabs/RB-000-Fault-Proofs-Recovery-Runbook-8dad0f1e6d4644c281b0e946c89f345f).
+
+### Unknown Absolute Prestate
+
+This absolute prestate is essentially a hash of the op-program memory loaded by Cannon, including its code.
+As such, the absolute prestate (or simple prestate) encodes the op-program. The MT-Cannon VM loads the op-program differently from single threaded Cannon.
+This warrants a new absolute prestate _hash_ to properly configure the `FaultDisputeGame` implementations for the MT-Cannon VM.
+
+- **Description:** This is when there is not a known prestate (preimage) for a given absolute prestate hash in the dispute game implementations.
+- **Risk Assessment:** High severity, Low likelihood.
+- **Mitigations:** Every absolute prestate is built off of an op-program release tag. The prestate is [build is reproducible](https://github.com/ethereum-optimism/optimism/blob/68f77aaa317b9184cbbcd1526bc57bce1722906b/op-program/Dockerfile.repro) such that the same prestate is emitted regardless of the environment.
+Furthermore, governance and Guardian signers will be instructed to reproduce the prestate build themselves and check that the prestate hash matches the op-program release that will be referenced in the MT-Cannon governance post.
```

Comment:
What do you think about a dedicated validations file in superchain-ops with generic instructions for this, so it can be referenced by all playbooks with prestate changes? Similar to how https://github.com/ethereum-optimism/superchain-ops/blob/main/NESTED-VALIDATION.md applies to all nested safe playbooks

Reply:
I'll add an AI to add instructions for prestate validation to the superchain-ops repo.

Reply:
I have suggested the AI with @Inphi has a DRI but could be changed with @pauldowman if needed [here](https://github.com/ethereum-optimism/design-docs/pull/123/files#r1951521083)

Reply:
It's a great idea, but do we need to block this FMA on that? I think we should prioritize it among all the other things we're currently doing. I don't personally have time to do this in the next few days, and @Inphi definitely can't, he's focusing on interop which is our top priority right now and taking > 100% effort.

Reply:
Oh yes! I just want to add this in action item section below to track in the future. This is indeed a non blocker for the FMA but at least we can track action to not forget this in the future.